test/parallel/test-repl-tab-complete.js

```diff
@@ -242,3 +244,16 @@ testMe.complete('cus', function(error, data) {
   completionCount++;
   assert.deepEqual(data, [['custom'], 'cus']);
 });
+
+// Make sure tab completion doesn't crash REPL with proxy objects
+putIn.run(['.clear']);
+
+putIn.run([
+  'var proxy = Proxy.create({});'
+]);
+
+testMe.complete('proxy.', function(error, data) {
```

Comment:
@cjihrig Can you PTAL at https://github.com/nodejs/io.js/pull/2122? I PRed that idea.